### DISCUSSION TITLE
STX: Reevaluate color balancing

### DISCUSSION THREAD

**Discussion by wrmacrae** [Senryoku/Draftmancer#452] [General]
In Strixhaven on Arena, packs do not have a mono-colored common of each color. For example, this pack does not have a white common: https://www.17lands.com/draft/e26fb1762c9c4f2b822fbd6141e22b2c/2/1

I'm not sure what if any color balancing rules _are_ running, though. From a handful of packs it seems like it might be true that for each color there is at least a gold card of that color, though I'm not positive if that counts hybrids, non-hybrids, and campuses, or just a subset of that, and I'm not sure if the lessons count.

I'm also not sure the overall best solution here given how the Color Balance option is mostly separated from the set in the UI. Maybe Color Balance should always be on by default, but should be implemented specially for certain sets. Or maybe a more elegant fix would be to "count" gold commons toward their colors, since we only see gold commons in guild/college sets.

If we want to count gold cards, that leaves two major questions to me:

1. Do lands count (using their color identity)? My guess is yes if they aren't in a land slot.
2. How do we implement collation to follow this rule without under- or over-representing certain cards? One approach would be to generate a list of random commons ignoring color, check for color representation, and discard the entire list if a color is missing; this slightly over-represents gold cards. I can think of other approaches, but I think they involve a lot of normalization math and we'd be more likely to make an error that makes card frequency skewed (like the old bug with common artifacts, https://github.com/Senryoku/MTGADraft/issues/139)

**Comment by wrmacrae** [comment]
This pack shows that hybrids count for their color, U in this case (assuming guild sets have any color balancing rules): https://www.17lands.com/draft/8916efcd55a64a2fb820665316df97e1/2/1
This pack shows that golds count for their color, W in this case (under the same assumption): https://www.17lands.com/draft/74e90a156ea14bd78608de08501a3f93/2/1
I'm going to work under the assumption that there is color balancing and that lands count because apparently lockets counted in GRN/RNA. I'm looking into methods that keep uniform card frequency, ensure a card of each color (counting golds/hybrids/lands), and doesn't force any particularly noticeable constraints or correlation beyond that.

**Comment by tooomm** [comment]
>lands count because apparently lockets counted in GRN/RNA

Why are you so sure lands (I assume you mean **non-basic lands** like dual lands?) count towards a color balancing when your reasoning are the special guild artifacts from GRN/RNA? There are non-basic lands (Guildgates) in GRN as well. Do they count?
https://mtg.fandom.com/wiki/Guilds_of_Ravnica#Cycles


Could it actually happen that there are no cards of a certain color (also no dfc, split, gold, hybrid cards etc. with that color in their mv/cmc), but a common dual land is in the pack as representation of that color?
That sounds unexpected to me, but I don't have the numbers and am not sure how color balancing exactly works.

**Comment by wrmacrae** [comment]
> Why are you so sure lands (I assume you mean non-basic lands like dual lands?) count towards a color balancing when your reasoning are the special guild artifacts from GRN/RNA?

Oh, to be clear, I'm not particularly sure about lands in STX, this is a wild guess because we have to pick and implement something, but I'd put it at ~60% likely. If we could find an Arena pack like the ones I linked that have a campus as the only common of one of its colors, then we'd have conclusive evidence they count; if we look at a lot of packs and find no such pack eventually we should think they do not count as their colors.

I think the case of guild gates and basic lands is very distinct from campuses in STX. Those lands go in their own slot, and just as rares and uncommons don't count for color balancing, I've been guessing special slots of all kinds (lessons, lands) do not count. This has a clear parallel to paper collation, because the sheets that have, for example, alternating BG commons could include a gold card or an appropriate land, whereas there's no clear way to make special slots count in paper (Arena collation rules are different, but I think they intend to imitate paper in certain respects). Campuses are a part of the normal pool of commons, which means they might act differently. I'm making the connection to lockets because both artifacts and lands have color identity but not color in these cases, but it's not a perfect link.

I just learned 17Lands now has public STX data available, so I'm gonna open a new issue tracking the work to grab that data and base packs on actual Arena packs when Ignore Collections is on and default settings are used. For future sets I imagine we can continue to lean on this public data, so these collation details will be important to keep unravelling in order to have the most representative experience between spoiler season and Arena release, but when we're this deep into a set we shouldn't have to use them anyway!

**Comment by wrmacrae** [comment]
Ah, so according to https://www.lethe.xyz/mtg/collation/index.html (which I had forgotten about because I can't access it from work with the suspicious TLD), in paper STX boosters actually _do_ have a mono-colored card of each color as we currently have, which makes the current implementation feel less wrong. I'd still like to investigate counting gold cards given what we've seen in Arena, but think #279 will be more valuable and important given this.

**Comment by wrmacrae** [comment]
Noting that SNC is not color balanced in Arena, or is color balanced differently in a way that counts golds (pack with no black common https://www.17lands.com/draft/6e96b6e748e14192b2534259e4b0ffa0/1/3), but SNC is color balanced in paper as STX was. Again this probably means it's not worth a change: https://www.lethe.xyz/mtg/collation/snc.html
